controllers/testutil/equalities.go

```diff
@@ -119,7 +120,7 @@ func deploymentsAreEqual(expected, actual *unstructured.Unstructured) (bool, err
 		for i, expectedEnv := range expectedContainer.Env {
 			actualEnv := actualContainer.Env[i]
 			if !reflect.DeepEqual(expectedEnv, actualEnv) {
-				return false, notEqualMsg(fmt.Sprintf("Container Env [expected: %s, actual: %s]", expectedEnv.Name, actualEnv.Name))
+				return false, notEqualMsg(fmt.Sprintf("Container Env [expected: %s=%s, actual: %s=%s]", expectedEnv.Name, expectedEnv.Value, actualEnv.Name, actualEnv.Value))
```

Comment:
`EnvVar.Value` is an optional field, sometimes it can be `ValueFrom` so if you want to show the value I think you need to check for the case that it's not `""`

Reply:
https://pkg.go.dev/k8s.io/api@v0.27.2/core/v1#EnvVar

Reply:
if it's default value is "", wouldn't this just print something like
`Container Env [expected: foo=bar, actual: foo=]`

imo that's fine and not worth complicating this block with conditionals just for logging

Reply:
yeah it would, I think it's a bit misleading if the actual is "ValueFrom", in such a case the expectedEnv.Value may also be different (empty if we're testing valuefrom), but I don't feel strong about this, just pointing it out.